core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -78,18 +76,25 @@
 
   /** If the stream is closed, this can only go from false to true. */
   protected boolean mClosed;
-  /** Whether or not the current block should be cached. */
-  protected boolean mShouldCacheCurrentBlock;
-  /** Include incomplete blocks if Alluxio is configured to store blocks in Alluxio storage. */
-  private final boolean mShouldCachePartiallyReadBlock;
- /** Current position of the stream. */
+  /**
+   * Current position of the file instream.
+   */
   protected long mPos;
+
+  /**
+   * Include partially read blocks if Alluxio is configured to cache blocks read blocks from ufs.
+   */
+  private final boolean mShouldCachePartiallyReadBlock;
```

Comment:
My understanding of this flag is: If we cache any part of a block (from ufs or remote read), we will cache the entire block, and it is a per-stream setting not system wide. 
Maybe `Cache the entire block even if only a portion of the block is read. Only valid when mShouldCache is true.`?